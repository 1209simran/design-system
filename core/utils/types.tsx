export type BaseProps = {
  className?: string,
  'data-test'?: string,
};

const baseProps = [
  'className',
  'data-test'
];

export const extractBaseProps = (props: Record<string, any>) => {
  const basePropsObj = baseProps.reduce((acc, curr) => {
    return (
      props[curr] ? { ...acc, [curr]: props[curr] } : { ...acc }
    );
  }, {});

  return basePropsObj;
};
