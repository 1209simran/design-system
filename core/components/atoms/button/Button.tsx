import * as React from 'react';
import Spinner from '@/components/atoms/spinner';
import classNames from 'classnames';

type ReactMouseEvent = (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;

export type Appearance = 'basic' | 'primary' | 'success' | 'alert' | 'transparent';

export type Size = 'tiny' | 'regular' | 'large';

export type Alignment = 'left' | 'right';

export interface IButtonProps {
  size?: Size;
  appearance?: Appearance;
  disabled?: boolean;
  expanded?: boolean;
  loading?: boolean;
  icon?: string;
  iconAlign?: Alignment;
  children?: string;
  onClick?: ReactMouseEvent;
  onMouseEnter?: ReactMouseEvent;
  onMouseLeave?: ReactMouseEvent;
}

const Button: React.FunctionComponent<IButtonProps> = props => {
  const {
    appearance = 'basic',
    size = 'regular',
    iconAlign = 'left',
    children,
    icon,
    expanded,
    loading,
    disabled,
    ...rest
  } = props;

  const buttonClass = classNames({
    ['Button']: true,
    ['Button--expanded']: expanded,
    [`Button--${size}`]: size,
    ['Button--square']: !children,
    [`Button--${appearance}`]: appearance,
    [`Button--iconAlign-${iconAlign}`]: children && iconAlign,
  });

  const iconClass = classNames({
    ['material-icons']: true,
    ['Button-icon']: true,
    [`Button-icon--${iconAlign}`]: children && iconAlign
  });

  return (
    <button className={buttonClass} disabled={disabled || loading} {...rest} >
      {loading && <Spinner size="small" appearance={(appearance === 'basic' || appearance === 'transparent') ? 'secondary' : 'white'} />}
      {icon && !loading && <i className={iconClass}>{icon}</i>}
      {children && `${children.charAt(0).toUpperCase()}${children.slice(1)}`}
    </button>
  );
};

Button.displayName = 'Button';

export default Button;
