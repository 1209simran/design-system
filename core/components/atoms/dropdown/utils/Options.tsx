export let dropdownOptions: any[] = [];
export const group1: any[] = [];
const group2: any[] = [];

for (let i = 1; i <= 40; i++) {
  group1.push({
    label: `Option ${i}`,
    value: `Option ${i}`,
  });
}

for (let i = 41; i <= 50; i++) {
  group2.push({
    label: `Option ${i}`,
    value: `Option ${i}`,
  });
}

dropdownOptions.push({
  group: true,
  label: 'Group 1',
  items: group1,
});

dropdownOptions.push({
  group: true,
  label: 'Group 2',
  items: group2,
});

export const selectedStoryOptions: any[] = [];

for (let i = 1; i <= 10; i++) {
  selectedStoryOptions.push({
    label: `Option ${i}`,
    value: `Option ${i}`,
    selected: i === 2,
  });
}

export const multiSelectedStoryOptions: any[] = [];

for (let i = 1; i <= 10; i++) {
  multiSelectedStoryOptions.push({
    label: `Option ${i}`,
    value: `Option ${i}`,
    selected: i === 2 || i === 7,
  });
}

export const storyOptions = group1.slice(0, 10);

export const storyWrapOptions = [
  {
    label: 'Design System Dropdown',
    value: 'Design System Dropdown'
  },
  {
    label: 'UI Kit Dropdown',
    value: 'UI Kit Dropdown',
  },
  {
    label: 'Innovaccer Analytics',
    value: 'Innovaccer Analytics'
  }
];

export const subInfoItems = [
  {
    label: 'Option 1',
    value: 'Option 1',
    subInfo: 'subInfo'
  },
  {
    label: 'Option 2',
    value: 'Option 2',
    subInfo: 'subInfo'
  },
  {
    label: 'Option 3',
    value: 'Option 3',
    subInfo: 'subInfo',
  },
];

export const iconItems = [
  {
    label: 'Option 1',
    value: 'Option 1',
    icon: 'events'
  },
  {
    label: 'Option 2',
    value: 'Option 2',
    icon: 'events',
  },
  {
    label: 'Option 3',
    value: 'Option 3',
    icon: 'events',
  },
];
