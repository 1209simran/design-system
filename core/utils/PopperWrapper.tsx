import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { Manager, Reference, Popper } from 'react-popper';

type PositionType =
  | 'auto-start'
  | 'auto'
  | 'auto-end'
  | 'top-start'
  | 'top'
  | 'top-end'
  | 'right-start'
  | 'right'
  | 'right-end'
  | 'bottom-end'
  | 'bottom'
  | 'bottom-start'
  | 'left-end'
  | 'left'
  | 'left-start'
  | undefined;

type actionType = 'click' | 'hover';

enum Offsets {
  Small = '2px',
  Medium = '4px',
  Large = '8px'
}

interface IProps {
  trigger: React.ReactElement<any>;
  placement: PositionType;
  children: React.ReactElement<any>;
  style: React.CSSProperties;
  appendToBody: boolean;
  on?: actionType;
  offset: keyof typeof Offsets;
  closeOnBackdropClick?: boolean;
  hoverable?: boolean;
}

interface IState {
  open: boolean;
  mouseLeaveDelay: number;
  mouseEnterDelay: number;
}

class PopperWrapper extends React.Component<IProps, IState> {
  private triggerRef: React.RefObject<HTMLElement>;
  private popupRef: React.RefObject<HTMLDivElement>;
  private _timer?: number;

  constructor(props: IProps) {
    super(props);

    this.state = {
      open: false,
      mouseLeaveDelay: 50,
      mouseEnterDelay: 0
    };

    this.triggerRef = React.createRef();
    this.popupRef = React.createRef();
  }

  public componentWillUnmount() {
    clearTimeout(this._timer);
    document.removeEventListener('mousedown', this.doesNodeContainClick);
  }

  public handleMouseLeave = (event: React.MouseEvent<HTMLElement>) => {
    const { hoverable = false } = this.props;

    if (hoverable) {
      clearTimeout(this._timer);
      this._timer = window.setTimeout(() => {
        this.setState({ open: false });
        if (this.props.children.props.onMouseLeave) {
          this.props.children.props.onMouseLeave(event);
        }
      }, this.state.mouseLeaveDelay);
    } else {
      this.setState({ open: false });
      if (this.props.children.props.onMouseLeave) {
        this.props.children.props.onMouseLeave(event);
      }
    }
  }

  public handleMouseEnter = (event: React.MouseEvent<HTMLElement>) => {
    const { hoverable = false } = this.props;

    if (hoverable) {
      clearTimeout(this._timer);
      this._timer = window.setTimeout(() => {
        this.setState({ open: true });
        if (this.props.children.props.onMouseEnter) {
          this.props.children.props.onMouseEnter(event);
        }
      }, this.state.mouseEnterDelay);
    } else {
      this.setState({ open: true });
      if (this.props.children.props.onMouseEnter) {
        this.props.children.props.onMouseEnter(event);
      }
    }
  }

  public togglePopper = () => {
    this.setState({ open: !this.state.open });
  }

  public doesNodeContainClick = (event: Event) => {
    if (
      !(
        this.findDOMNode(this.popupRef).contains(event.target as HTMLElement) ||
        this.findDOMNode(this.triggerRef).contains(event.target as HTMLElement)
      )
    ) {
      this.togglePopper();
    }
  }

  public componentDidMount() {
    const { on = 'click', closeOnBackdropClick = true } = this.props;
    const { open } = this.state;

    if (on === 'click' && open && closeOnBackdropClick) {
      document.addEventListener('mousedown', this.doesNodeContainClick);
    }
  }

  public componentDidUpdate() {
    const { on = 'click', closeOnBackdropClick = true } = this.props;
    const { open } = this.state;

    if (on === 'click' && open && closeOnBackdropClick) {
      document.addEventListener('mousedown', this.doesNodeContainClick);
    } else if (on === 'click' && !open && closeOnBackdropClick) {
      document.removeEventListener('mousedown', this.doesNodeContainClick);
    }
  }

  public getTriggerElement(trigger: React.ReactElement<any>, ref: React.Ref<any>, on: actionType) {
    const options =
      on === 'hover'
        ? {
          ref,
          onMouseEnter: this.handleMouseEnter,
          onMouseLeave: this.handleMouseLeave
        }
        : {
          ref,
          onClick: this.togglePopper
        };

    const element = React.cloneElement(
      (
        <span className="d-inline-block">
          {trigger}
        </span>
      ),
      options
      );

    return element;
  }

  public getChildrenElement(
    children: React.ReactElement<any>,
    ref: React.Ref<any>,
    placement: string,
    style: React.CSSProperties
  ) {
    const options = this.props.on === 'hover' ? {
      ref,
      style,
      onMouseEnter: this.handleMouseEnter,
      onMouseLeave: this.handleMouseLeave,
      'data-placement': placement
    } : {
      ref,
      style,
      'data-placement': placement
    };

    const element = React.cloneElement(children, options);
    return element;
  }

  public render() {
    const { trigger, children, placement, appendToBody, on = 'click', offset } = this.props;
    const { open } = this.state;

    return (
      <Manager>
        <Reference innerRef={this.triggerRef}>{({ ref }) => this.getTriggerElement(trigger, ref, on)}</Reference>
        {open &&
          appendToBody &&
          ReactDOM.createPortal(
            (
              /* tslint:disable:no-shadowed-variable */
              <Popper placement={placement} innerRef={this.popupRef}>
                {({ ref, style, placement }) => {
                  const newStyle = offset ? this.getUpdatedStyle(style, placement, offset) : style;
                  return this.getChildrenElement(children, ref, placement, newStyle);
                }}
              </Popper>
            ),
            document.body
          )}
        {open && !appendToBody && (
          <Popper placement={placement} innerRef={this.popupRef}>
            {({ ref, style, placement }) => {
              const newStyle = offset ? this.getUpdatedStyle(style, placement, offset) : style;
              return this.getChildrenElement(children, ref, placement, newStyle);
            }}
          </Popper>
        )}
      </Manager>
    );
  }

  private getUpdatedStyle = (
    oldStyle: React.CSSProperties,
    placement: PositionType,
    offset: keyof typeof Offsets = 'Medium'
  ) => {
    const { style = {} } = this.props;
    const newStyle = { ...style, ...oldStyle };
    const position = placement ? placement.split('-')[0] : placement;
    switch (position) {
      case 'top':
        newStyle.marginBottom = Offsets[offset];
        break;

      case 'bottom':
        newStyle.marginTop = Offsets[offset];
        break;

      case 'left':
        newStyle.marginRight = Offsets[offset];
        break;

      case 'right':
        newStyle.marginLeft = Offsets[offset];
        break;
    }
    return newStyle;
  }

  private findDOMNode = (ref: React.RefObject<HTMLElement>) => {
    return ReactDOM.findDOMNode(ref.current!) as HTMLElement;
  }
}

export default PopperWrapper;
