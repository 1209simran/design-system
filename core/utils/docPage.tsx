// @ts-nocheck

import * as React from 'react';
import { Title, Props, Description, Primary, Story, PropsProps } from '@storybook/addon-docs/blocks';

interface DocPageProps {
  title?: string;
  description?: React.ReactNode;
  props?: PropsProps;
}

export const docPage = (props: DocPageProps) => {
  const {
    title,
    description,
    props,
    stories
  } = props;

  const separatorIndex = title?.indexOf('|');

  return (
    <>
      <Title>{title && separatorIndex !== -1 ? title.slice(separatorIndex + 1) : title}</Title>
      <Description>
        {description}
      </Description>
      {/* <Primary /> */}
      {/* <Story id="switch-state--alert"  /> */}
      <Props {...props} />
    </>
  );
};

export default docPage;
