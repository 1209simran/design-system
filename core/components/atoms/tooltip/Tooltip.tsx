import * as React from 'react';
import { Omit } from 'utility-types';
import { PopperWrapper } from '@/utils';

type PositionType = 'top' | 'bottom' | 'left' | 'right';
type IDivProps = Omit<JSX.IntrinsicElements['div'], 'ref'>;

export interface ITooltipProps extends IDivProps {
  tooltip: string;
  children: React.ReactElement<any>;
  position?: PositionType;
  appendToBody?: boolean;
}

interface IState {
  position: {
    top: number;
    left: number;
  };
  style: React.CSSProperties;
}

/**
 * Tooltip is used to displays floating content in relation to a target when that target is hovered.
 *
 * Tooltips mostly appear either at the top or bottom of their target.
 * The preferred and default side is the bottom.
 *
 * For left navigation with only icons, show tooltip on the right.
 * @class Tooltip
 * @extends {React.Component<ITooltipProps, IState>}
 */
class Tooltip extends React.Component<ITooltipProps, IState> {
  constructor(props: ITooltipProps) {
    super(props);

    this.state = {
      position: {
        top: 0,
        left: 0,
      },
      style: {}
    };
  }

  public static defaultProps: Partial<ITooltipProps> = {
    position: 'bottom',
    appendToBody: true
  };

  public render() {
    const {
      appendToBody = false,
      tooltip,
      children,
      position = '',
      ...props
    } = this.props;

    const tooltipWrapper = (
      <div
        className="Tooltip"
        {...props}
        style={this.state.style}
      >
        {tooltip}
      </div>
    );

    return (
      <PopperWrapper
        trigger={children}
        placement={this.props.position}
        style={this.state.style}
        appendToBody={appendToBody}
        on={'hover'}
        offset={'Medium'}
      >
        {tooltipWrapper}
      </PopperWrapper>
    );
  }
}

export default Tooltip;
