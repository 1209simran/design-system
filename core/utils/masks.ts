import { Mask } from '@/utils/types';
import { DatePickerProps } from "@/index.type";

type DateFormat = DatePickerProps['inputFormat'];
export const date: Record<DateFormat, Mask> = {
  'dd/mm/yyyy': [/[0123]/, /\d/, '/', /\[01]/, /\d/, '/', /\d/, /\d/, /\d/, /\d/],
  'mm/dd/yyyy': [/[01]/, /\d/, '/', /[0123]/, /\d/, '/', /\d/, /\d/, /\d/, /\d/],
  'yyyy/mm/dd': [/\d/, /\d/, /\d/, /\d/, '/', /[01]/, /\d/, '/', /[0123]/, /\d/],
  'dd-mm-yyyy': [/[0123]/, /\d/, '-', /[01]/, /\d/, '-', /\d/, /\d/, /\d/, /\d/],
  'mm-dd-yyyy': [/[01]/, /\d/, '-', /[0123]/, /\d/, '-', /\d/, /\d/, /\d/, /\d/],
  'yyyy-mm-dd': [/\d/, /\d/, /\d/, /\d/, '-', /[01]/, /\d/, '-', /[0123]/, /\d/],
};
