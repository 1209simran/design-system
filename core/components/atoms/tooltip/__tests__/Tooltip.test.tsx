import * as React from 'react';
import { shallow } from 'enzyme';
import Tooltip, { ITooltipProps as IProps } from '../Tooltip';
import Button from '@/components/atoms/button';
import { TestHelper, filterUndefined, valueHelper, testMessageHelper } from '@/utils/TestHelper';

const Position = ['top', 'bottom', 'left', 'right'];
const StringValue = 'Sample string';
const BooleanValue = [true, false];
const ReactNodeValue = <Button>Button</Button>;

const Mapper = {
  position: valueHelper(Position, { iterate: true }),
  tooltip: valueHelper(StringValue, { required: true }),
  children: valueHelper(ReactNodeValue, { required: true }),
  appendToBody: valueHelper(BooleanValue, { iterate: true }),
};

describe('Tooltip component', () => {
  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as IProps;
    const { children, ...rest } = attr;

    it(testMessageHelper(attr), () => {
      const tree = shallow(
        <Tooltip
          {...rest}
        >
          {children}
        </Tooltip>
      );
      expect(tree).toMatchSnapshot();
    });
  };

  TestHelper(Mapper, testFunc);
});
