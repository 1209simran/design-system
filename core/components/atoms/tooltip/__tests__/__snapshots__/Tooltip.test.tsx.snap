// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip component 
  tooltip: "Sample string", children: "[ReactNode]"
 1`] = `
<PopperWrapper
  appendToBody={true}
  offset="Medium"
  on="hover"
  placement="bottom"
  style={Object {}}
  trigger={
    <Button>
      Button
    </Button>
  }
>
  <div
    className="Tooltip"
    style={Object {}}
  >
    Sample string
  </div>
</PopperWrapper>
`;

exports[`Tooltip component 
  tooltip: "Sample string", children: "[ReactNode]", appendToBody: false
 1`] = `
<PopperWrapper
  appendToBody={false}
  offset="Medium"
  on="hover"
  placement="bottom"
  style={Object {}}
  trigger={
    <Button>
      Button
    </Button>
  }
>
  <div
    className="Tooltip"
    style={Object {}}
  >
    Sample string
  </div>
</PopperWrapper>
`;

exports[`Tooltip component 
  tooltip: "Sample string", children: "[ReactNode]", appendToBody: true
 1`] = `
<PopperWrapper
  appendToBody={true}
  offset="Medium"
  on="hover"
  placement="bottom"
  style={Object {}}
  trigger={
    <Button>
      Button
    </Button>
  }
>
  <div
    className="Tooltip"
    style={Object {}}
  >
    Sample string
  </div>
</PopperWrapper>
`;

exports[`Tooltip component 
  tooltip: "Sample string", children: "[ReactNode]", position: "bottom"
 1`] = `
<PopperWrapper
  appendToBody={true}
  offset="Medium"
  on="hover"
  placement="bottom"
  style={Object {}}
  trigger={
    <Button>
      Button
    </Button>
  }
>
  <div
    className="Tooltip"
    style={Object {}}
  >
    Sample string
  </div>
</PopperWrapper>
`;

exports[`Tooltip component 
  tooltip: "Sample string", children: "[ReactNode]", position: "bottom", appendToBody: false
 1`] = `
<PopperWrapper
  appendToBody={false}
  offset="Medium"
  on="hover"
  placement="bottom"
  style={Object {}}
  trigger={
    <Button>
      Button
    </Button>
  }
>
  <div
    className="Tooltip"
    style={Object {}}
  >
    Sample string
  </div>
</PopperWrapper>
`;

exports[`Tooltip component 
  tooltip: "Sample string", children: "[ReactNode]", position: "bottom", appendToBody: true
 1`] = `
<PopperWrapper
  appendToBody={true}
  offset="Medium"
  on="hover"
  placement="bottom"
  style={Object {}}
  trigger={
    <Button>
      Button
    </Button>
  }
>
  <div
    className="Tooltip"
    style={Object {}}
  >
    Sample string
  </div>
</PopperWrapper>
`;

exports[`Tooltip component 
  tooltip: "Sample string", children: "[ReactNode]", position: "left"
 1`] = `
<PopperWrapper
  appendToBody={true}
  offset="Medium"
  on="hover"
  placement="left"
  style={Object {}}
  trigger={
    <Button>
      Button
    </Button>
  }
>
  <div
    className="Tooltip"
    style={Object {}}
  >
    Sample string
  </div>
</PopperWrapper>
`;

exports[`Tooltip component 
  tooltip: "Sample string", children: "[ReactNode]", position: "left", appendToBody: false
 1`] = `
<PopperWrapper
  appendToBody={false}
  offset="Medium"
  on="hover"
  placement="left"
  style={Object {}}
  trigger={
    <Button>
      Button
    </Button>
  }
>
  <div
    className="Tooltip"
    style={Object {}}
  >
    Sample string
  </div>
</PopperWrapper>
`;

exports[`Tooltip component 
  tooltip: "Sample string", children: "[ReactNode]", position: "left", appendToBody: true
 1`] = `
<PopperWrapper
  appendToBody={true}
  offset="Medium"
  on="hover"
  placement="left"
  style={Object {}}
  trigger={
    <Button>
      Button
    </Button>
  }
>
  <div
    className="Tooltip"
    style={Object {}}
  >
    Sample string
  </div>
</PopperWrapper>
`;

exports[`Tooltip component 
  tooltip: "Sample string", children: "[ReactNode]", position: "right"
 1`] = `
<PopperWrapper
  appendToBody={true}
  offset="Medium"
  on="hover"
  placement="right"
  style={Object {}}
  trigger={
    <Button>
      Button
    </Button>
  }
>
  <div
    className="Tooltip"
    style={Object {}}
  >
    Sample string
  </div>
</PopperWrapper>
`;

exports[`Tooltip component 
  tooltip: "Sample string", children: "[ReactNode]", position: "right", appendToBody: false
 1`] = `
<PopperWrapper
  appendToBody={false}
  offset="Medium"
  on="hover"
  placement="right"
  style={Object {}}
  trigger={
    <Button>
      Button
    </Button>
  }
>
  <div
    className="Tooltip"
    style={Object {}}
  >
    Sample string
  </div>
</PopperWrapper>
`;

exports[`Tooltip component 
  tooltip: "Sample string", children: "[ReactNode]", position: "right", appendToBody: true
 1`] = `
<PopperWrapper
  appendToBody={true}
  offset="Medium"
  on="hover"
  placement="right"
  style={Object {}}
  trigger={
    <Button>
      Button
    </Button>
  }
>
  <div
    className="Tooltip"
    style={Object {}}
  >
    Sample string
  </div>
</PopperWrapper>
`;

exports[`Tooltip component 
  tooltip: "Sample string", children: "[ReactNode]", position: "top"
 1`] = `
<PopperWrapper
  appendToBody={true}
  offset="Medium"
  on="hover"
  placement="top"
  style={Object {}}
  trigger={
    <Button>
      Button
    </Button>
  }
>
  <div
    className="Tooltip"
    style={Object {}}
  >
    Sample string
  </div>
</PopperWrapper>
`;

exports[`Tooltip component 
  tooltip: "Sample string", children: "[ReactNode]", position: "top", appendToBody: false
 1`] = `
<PopperWrapper
  appendToBody={false}
  offset="Medium"
  on="hover"
  placement="top"
  style={Object {}}
  trigger={
    <Button>
      Button
    </Button>
  }
>
  <div
    className="Tooltip"
    style={Object {}}
  >
    Sample string
  </div>
</PopperWrapper>
`;

exports[`Tooltip component 
  tooltip: "Sample string", children: "[ReactNode]", position: "top", appendToBody: true
 1`] = `
<PopperWrapper
  appendToBody={true}
  offset="Medium"
  on="hover"
  placement="top"
  style={Object {}}
  trigger={
    <Button>
      Button
    </Button>
  }
>
  <div
    className="Tooltip"
    style={Object {}}
  >
    Sample string
  </div>
</PopperWrapper>
`;
